enhancements/results-forwarding.md

```diff
@@ -0,0 +1,523 @@
+recommendations if the supplied forwarding endpoint is valid. The Compliance
+Operator will also continue to use CRDs and PVCs to store results,
+remediations, and evidence as it does today.
+
+A user may explicitly disable in-cluster storage if forwarding is enabled. The
+Compliance Operator will not support disabling PVC storage without a valid
+forwarding endpoint.
+
+This will require changing the **aggregator** to always forward via a gRPC API
+to a configured implementation. In this case, the **aggregator** will act as a
+gRPC client. The **resultserver** then becomes a gRPC server, subject to the
+evidence forwarding provider and will be renamed **evidence-persistor** and
+responsible for writing results as CRDs. Alternatively, the **aggregator** may
+forward results to an external system that uses the same gRPC API. This allows
+users the ability to offload compliance results from the cluster.
+
+Forwarder implementations would be configured via the `ScanSetting` and they'd
+be called "providers". For backwards compatibility, we'd have defaults that
+would point to the current mode of operation of the Compliance Operator (e.g.,
+a `pvc` provider for evidence storage.)
+
+For more detail on the proposed API, see the *API Extensions* section.
+
+### User Stories
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to
+  offload compliance result storage to a dedicated system, instead of having
+  each cluster own its own compliance data.
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to have
+  one place to view the compliance stance of my whole fleet.
+
+### API Extensions
+
+Given the scope of the change, we'd upgrade the `ScanSetting` version to
+`v1alpha2`. The following sections describe the additional values needed by the
+`ScanSetting` CRD.
+
+#### `resultForwarding.provider`
+
+Required: True
+Type: string
+
+The implementation to use for forwarding. The initial implementation will
+support `grpc`.
+
+#### `resultForwarding.grpc.endpoint`
+
+Required: True
+Type: string
+
+Endpoint URL to a gRPC server for result storage. The service may be in or
+outside the cluster. One potential implementation may be
+[compserv](https://github.com/rhmdnd/compserv).
+
+#### `resultForwarding.grpc.version`
+
+Required: False
+Default: `v1`
+Type: string
+
+#### `resultForwarding.grpc.tls.dynamic.caSecretName`
+
+Required: True
+Type: string
+
+Reference to a CA certificate and key pair for TLS.
+
+#### `resultForwarding.grpc.tls.dynamic.serverSecretName`
+
+Required: True
+Type: string
+
+Reference to a server certificate/keypair secret for mutual TLS with the
+forwarding endpoint.
+
+#### `resultForwarding.grpc.tls.dynamic.clientSecretName`
+
+Required: True if using mutual TLS
+Type: string
+
+Reference to a client certificate/keypair secret for mutual TLS.
+
+#### `resultForwarding.grpc.authentication.type`
+
+Required: False
+Default: `mtls`
+Type: string
+Choices: `mtls`, `token`
```

Comment:
It would be nice to infer this from the authentication config. I'm not sure if it's preferrable though. But I am thinking of an overall auth layout like:

For mTLS:
```
resultForwarding:
  provider: grpc
  grpc:
    version: v1
    endpoint: https://compliance-endpoint.example.com
    authentication:
      serverCA: $ConfigMapRef (optional)
      mtlsClientCert: $SecretRef
```
For token:
```
resultForwarding:
  provider: grpc
  grpc:
    version: v1
    endpoint: https://compliance-endpoint.example.com
    authentication:
      serverCA: $ConfigMapRef (optional)
      token: $SecretRef
```
* Specify either mtlsClientCert or token to select mtls/token
* No server cert is required for prepping the client end of mTLS - but you do still need the server's CA. So I've omitted the server cert and added serverCA.
* serverCA is an optional configmap providing a CA bundle to use to authenticate the server. Otherwise, with mTLS we use the CA provided in the client cert's PEM file for this.
* For token, serverCA would be used to authenticate the server's TLS connection. If serverCA is not provided with token, then we rely on the default CA trust.